libs/db/db-admin-impl/src/main/kotlin/net/corda/db/admin/impl/LiquibaseSchemaMigratorImpl.kt

```
@@ -122,12 +144,39 @@ class LiquibaseSchemaMigratorImpl(
                 database
             )
 
-            log.info("Updating ${database.databaseProductName} ${database.databaseProductVersion} " +
-                    "DB Schema for ${database.connection.catalog}")
-            if (null == sql) {
-                lb.update(tag, Contexts())
-            } else {
-                lb.update(tag, Contexts(), sql)
+            log.info(
+                "Updating ${database.databaseProductName} ${database.databaseProductVersion} " +
+                        "DB Schema for ${database.connection.catalog}"
+            )
+
+            try {
+                val scopeObjects = mapOf(
+                    Scope.Attr.database.name to lb.database,
+                    Scope.Attr.resourceAccessor.name to lb.resourceAccessor
+                )
+                Scope.child(scopeObjects) {
+                    if (sql != null) {
+                        commandScopeFactory(UpdateSqlCommandStep.COMMAND_NAME).configure(lb, tag).also {
+                            it.setOutput(
+                                WriterOutputStream(
+                                    sql,
+                                    GlobalConfiguration.OUTPUT_FILE_ENCODING.currentValue
+                                )
+                            )
+                            it.execute()
+                        }
+                    } else {
+                        commandScopeFactory(UpdateCommandStep.COMMAND_NAME).configure(lb, tag).also {
+                            it.execute()
+                        }
+                    }
+                }
+            } catch (e: Exception) {
+                if (e is LiquibaseException) {
+                    throw e
+                } else {
+                    throw LiquibaseException(e)
+                }
             }
```

I'm not sure this is the right thing to do. Why do we wrap everything in a LiquibaseException?

If we wanted a specific type of exception, I'd say it should be a custom exception because I don't think it should be tied to the 3rd party implementation.
If we're not worried about catching a specific type if exception, then I think we should just let it bubble up.